### PR TITLE
fix: Implement infinityLoader for event speakers

### PR DESCRIPTION
**Pull request by kushthedude**
Refers #3650

### CODE REVIEW

**Comment by kushthedude** [comment]
@iamareebjamal Let's include this in the latest release, I tested it on Summit 2020 results are really astonishing:
Before this: Speaker page loads in 20+ Seconds
After this: Speaker page loads in < 8 Seconds 

Also this 8 seconds will further be reduced to mere 3-4 seconds once the settings are removed from the global injection.

File: app/routes/public/speakers.js [hunk 1]
```diff
@@ -3,36 +3,40 @@ import Route from '@ember/routing/route';
 export default Route.extend({
   async model() {
     const eventDetails = this.modelFor('public');
-    return {
-      event    : eventDetails,
-      speakers : await eventDetails.query('speakers', {
-        filter: [
+    let filterOptions = [];
```
Comment (iamareebjamal) [inline app/routes/public/speakers.js]: Why is it let?

Comment (kushthedude) [reply app/routes/public/speakers.js]: Changing to const !

Comment (kushthedude) [reply app/routes/public/speakers.js]: @iamareebjamal Done

Comment (kushthedude) [reply app/routes/public/speakers.js]: Changing to const will fail the ESLintrule as filterOption may tend to change , hence let should be used

Comment (iamareebjamal) [reply app/routes/public/speakers.js]: > Changing to const will fail the ESLintrule

Then that ESLintrule is stupid. Nowhere in this code filterOptions is changed, neither can this reference ever be changed

Change it to const

**Comment by kushthedude** [comment]
Should I disable the rule or disable the rule in this file only?

**Comment by iamareebjamal** [comment]
What's the rule identifier?

**Comment by kushthedude** [comment]
> What's the rule identifier?

```sh
1095            7:5 - 'filterOptions' is constant. (no-const-assign)
```

**Comment by iamareebjamal** [comment]
That's triggered because you change to const and reassign in the next line. How is that ever going to work? :man_facepalming: const means no changing the value, do the declaration and assignment on the same line and change let to const

**Comment by kushthedude** [comment]
> That's triggered because you change to const and reassign in the next line. How is that ever going to work? 🤦‍♂ const means no changing the value, do the declaration and assignment on the same line and change let to const
> 
> https://eslint.org/docs/rules/no-const-assign

Done 😅

**Pull request merged**